pkg/kube/inject/webhook.go

```diff
@@ -608,10 +610,45 @@ func applyPrometheusMerge(pod *corev1.Pod, mesh *meshconfig.MeshConfig) error {
 		pod.Annotations["prometheus.io/port"] = targetPort
 		pod.Annotations["prometheus.io/path"] = "/stats/prometheus"
 		pod.Annotations["prometheus.io/scrape"] = "true"
+		return nil
 	}
+
 	return nil
 }
 
+var (
+	prometheusScrapeAnnotation = "prometheus_io_scrape"
+	prometheusPortAnnotation   = "prometheus_io_port"
+)
```

Comment:
what's this?

Reply:
prometheus will convert annotation to this format see [here](https://github.com/prometheus/prometheus/blob/71a0f42331566a8849863d77078083edbb0b3bc4/discovery/kubernetes/pod.go#L205)

`prometheus.io/scrape` `prometheus.io.scrape` `prometheus-io/scrape` have the same meaning in Prometheus

Reply:
Putting that comment in the code would be ideal for future reference